### ISSUE TITLE
Error recorded from infeed: Op type not registered 'ImageProjectiveTransformV2'

### ISSUE THREAD

**Issue by prashastk** [tensorflow/tpu#263]
I am facing this issue when trying to train an inception v3 model on TPU. I have taken the code from https://github.com/tensorflow/tpu/blob/master/models/experimental/inception/inception_v3.py and modified the input data pipeline.

I've narrowed down my usage to `tensorflow.contrib.image.rotate` that causes this error:
Error recorded from infeed: Op type not registered 'ImageProjectiveTransformV2' in binary running on n-72095697-w-0. Make sure the Op and Kernel are registered in the binary running in this process. Note that if you are loading a saved graph which used ops from tf.contrib, accessing (e.g.) `tf.contrib.resampler` should be done before importing the graph, as contrib ops are lazily registered when the module is first accessed.

If I don't use `tensorflow.contrib.image.rotate` the code works fine. How can I get around this issue? 

Full stack:
ERROR   2019-01-17 16:32:50 -0800       master-replica-0                I0118 00:32:50.893857 140074737563392 tf_logging.py:115] Enqueue next (100) batch(es) of data to infeed.
ERROR   2019-01-17 16:32:50 -0800       master-replica-0                I0118 00:32:50.894562 140074737563392 tf_logging.py:115] Dequeue next (100) batch(es) of data from outfeed.
INFO    2019-01-17 16:32:56 -0800       master-replica-0                Error recorded from infeed: Op type not registered 'ImageProjectiveTransformV2' in binary running on n-3a08239c-w-0. Make sure the Op and Kernel are registered in the binary running in this process. Note that if you are loading a saved graph which used ops from tf.contrib, accessing (e.g.) `tf.contrib.resampler` should be done before importing the graph, as contrib ops are lazily registered when the module is first accessed.
INFO    2019-01-17 16:32:56 -0800       master-replica-0                         [[node input_pipeline_task0/while/IteratorGetNext (defined at /usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py:3053)  = IteratorGetNext[_class=["loc:@input_pipeline_task0/while/InfeedQueue/split/1"], output_shapes=[[1024,299,299,3], [1024,7]], output_types=[DT_FLOAT, DT_INT32], _device="/job:worker/replica:0/task:0/device:CPU:0"](input_pipeline_task0/while/IteratorGetNext/Enter, ^input_pipeline_task0/while/Identity)]]
INFO    2019-01-17 16:32:56 -0800       master-replica-0                
INFO    2019-01-17 16:32:56 -0800       master-replica-0                Caused by op 'input_pipeline_task0/while/IteratorGetNext', defined at:
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/lib/python3.5/runpy.py", line 184, in _run_module_as_main
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    "__main__", mod_spec)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/lib/python3.5/runpy.py", line 85, in _run_code
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    exec(code, run_globals)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/root/.local/lib/python3.5/site-packages/trainer/tpu.py", line 633, in <module>
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    app.run(main)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/absl/app.py", line 300, in run
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    _run_main(main, args)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/absl/app.py", line 251, in _run_main
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    sys.exit(main(argv))
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/root/.local/lib/python3.5/site-packages/trainer/tpu.py", line 618, in main
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    input_fn=train_fn, steps=FLAGS.train_steps_per_eval)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 2403, in train
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    saving_listeners=saving_listeners
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/estimator/estimator.py", line 354, in train
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    loss = self._train_model(input_fn, hooks, saving_listeners)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/estimator/estimator.py", line 1207, in _train_model
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    return self._train_model_default(input_fn, hooks, saving_listeners)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/estimator/estimator.py", line 1237, in _train_model_default
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    features, labels, model_fn_lib.ModeKeys.TRAIN, self.config)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 2195, in _call_model_fn
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    features, labels, mode, config)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/estimator/estimator.py", line 1195, in _call_model_fn
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    model_fn_results = self._model_fn(features=features, **kwargs)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 2491, in _model_fn
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    input_holders.generate_infeed_enqueue_ops_and_dequeue_fn())
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 1103, in generate_infeed_enqueue_ops_and_dequeue_fn
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    self._invoke_input_fn_and_record_structure())
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 1207, in _invoke_input_fn_and_record_structure
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    wrap_fn(device=host_device, op_fn=enqueue_ops_fn))
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 2877, in _wrap_computation_in_while_loop
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    parallel_iterations=1)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/control_flow_ops.py", line 3291, in while_loop
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    return_same_structure)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/control_flow_ops.py", line 3004, in BuildLoop
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    pred, body, original_loop_vars, loop_vars, shape_invariants)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/control_flow_ops.py", line 2939, in _BuildLoop
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    body_result = body(*packed_vars_for_body)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 2866, in computation
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    with ops.control_dependencies(op_fn()):
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 725, in enqueue_ops_fn
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    features, labels = inputs.features_and_labels()
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 3053, in features_and_labels
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    return _Inputs._parse_inputs(self._iterator.get_next())
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/data/ops/iterator_ops.py", line 421, in get_next
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    name=name)), self._output_types,
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/ops/gen_dataset_ops.py", line 2069, in iterator_get_next
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    output_shapes=output_shapes, name=name)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    op_def=op_def)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/util/deprecation.py", line 488, in new_func
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    return func(*args, **kwargs)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/ops.py", line 3274, in create_op
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    op_def=op_def)
INFO    2019-01-17 16:32:56 -0800       master-replica-0                  File "/usr/local/lib/python3.5/dist-packages/tensorflow/python/framework/ops.py", line 1770, in __init__
INFO    2019-01-17 16:32:56 -0800       master-replica-0                    self._traceback = tf_stack.extract_stack()
INFO    2019-01-17 16:32:56 -0800       master-replica-0                
INFO    2019-01-17 16:32:56 -0800       master-replica-0                NotFoundError (see above for traceback): Op type not registered 'ImageProjectiveTransformV2' in binary running on n-3a08239c-w-0. Make sure the Op and Kernel are registered in the binary running in this process. Note that if you are loading a saved graph which used ops from tf.contrib, accessing (e.g.) `tf.contrib.resampler` should be done before importing the graph, as contrib ops are lazily registered when the module is first accessed.
INFO    2019-01-17 16:32:56 -0800       master-replica-0                         [[node input_pipeline_task0/while/IteratorGetNext (defined at /usr/local/lib/python3.5/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py:3053)  = IteratorGetNext[_class=["loc:@input_pipeline_task0/while/InfeedQueue/split/1"], output_shapes=[[1024,299,299,3], [1024,7]], output_types=[DT_FLOAT, DT_INT32], _device="/job:worker/replica:0/task:0/device:CPU:0"](input_pipeline_task0/while/IteratorGetNext/Enter, ^input_pipeline_task0/while/Identity)]]
ERROR   2019-01-17 16:32:56 -0800       master-replica-0                I0118 00:32:56.385142 140073059481344 tf_logging.py:115] Error recorded from infeed: Op type not registered 'ImageProjectiveTransformV2' in binary running on n-3a08239c-w-0. Make sure the Op and Kernel are registered in the binary running in this process. Note that if you are loading a saved graph which used ops from tf.contrib, accessing (e.g.) `tf.contrib.resampler` should be done before importing the graph, as contrib ops are lazily registered when the module is first accessed.

**Comment by idorozenberg** [comment]
Also getting the same error. I'm using the ml-engine to train tf object detection api.
I added the auto-augment code to my private repo and tried to train it via the instructions.

**Comment by kovtun77** [comment]
Importing image module explicitly solved the issue for me.
`from tensorflow.contrib import image`

**Comment by HangHuang** [comment]
Also getting the same error.Importing image module explicitly don't solve the issue for me.How can I get around this issue?
